llvm/include/llvm/Target/Target.td

```diff
@@ -1564,6 +1562,10 @@ class AsmParser {
   // method shall be called for all operands as opposed to only those
   // that have their own specified custom parsers.
   bit CallCustomParserForAllOperands = false;
+
+  // PreferSmallerInstructions - Should the assembly matcher prefer the smaller
+  // instructions.
```

Comment:
I would add to this comment

(a) that this flag is useful for Arm (so that people who only know other architectures have an idea why it needs to be here in the first place)

(b) what the preference order is if this flag is false. (Surely the only reason this flag exists, instead of just being always-true, is so that somebody can deliberately set it to false, in which case they have some reason to want a different preference order, and will need to know how to get it.)

Reply:
Addressing in https://github.com/llvm/llvm-project/pull/83587

Reply:
Yes I agree with @kosarev also. The ARM backend has a massive number of ambiguous matchables (some are exact duplicate aliases, some are meaningfully different) that we rely on the stable sort to not change the order of. 
I have a patch to remove a handful of completely redundant matchables I'll clean up and send upstream at some point. 

As discussed in https://github.com/llvm/llvm-project/pull/84483 I do think there is a case to order by predicates (similar to current sorting by register classes), however there isn't a good way to do this that I can see.